e2e_test/streaming/asof_join.slt

```diff
@@ -0,0 +1,143 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+# asof inner join
+
+statement ok
+create table t1 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+create table t2 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+create materialized view mv1 as SELECT t1.v1 t1_v1, t1.v2 t1_v2, t1.v3 t1_v3, t2.v1 t2_v1, t2.v2 t2_v2, t2.v3 t2_v3 FROM t1 ASOF JOIN t2 ON t1.v1 = t2.v1 and t1.v2 <= t2.v2;
+
+statement ok
+insert into t1 values (1, 2, 3);
+
+statement ok
+insert into t2 values (1, 3, 4);
+
+query III
+select * from mv1;
+----
+1 2 3 1 3 4
+
+statement ok
+insert into t2 values (1, 2, 3);
+
+query III
+select * from mv1;
+----
+1 2 3 1 2 3
+
+statement ok
+delete from t1 where v3 = 3;
+
+query III
+select * from mv1;
+----
+
+
+statement ok
+insert into t1 values (2, 3, 4);
+
+statement ok
+insert into t2 values (2, 3, 6), (2, 3, 7), (2, 3, 5);
+
+query III
+select * from mv1;
+----
+2 3 4 2 3 5
+
+statement ok
+insert into t2 values (2, 3, 1), (2, 3, 2);
+
+query III
+select * from mv1;
+----
+2 3 4 2 3 1
+
+statement ok
+drop materialized view mv1;
+
+statement ok
+drop table t1;
+
+statement ok
+drop table t2;
+
+
+# asof left join
+
+statement ok
+create table t1 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+create table t2 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+create materialized view mv1 as SELECT t1.v1 t1_v1, t1.v2 t1_v2, t1.v3 t1_v3, t2.v1 t2_v1, t2.v2 t2_v2, t2.v3 t2_v3 FROM t1 ASOF LEFT JOIN t2 ON t1.v1 = t2.v1 and t1.v2 > t2.v2;
```

Comment:
Suggest to use "LEFT ASOF JOIN" instead of `ASOF LEFT JOIN`, because clickhouse uses this syntax.

https://clickhouse.com/docs/en/sql-reference/statements/select/join#:~:text=ASOF%20JOIN%20and-,LEFT%20ASOF%20JOIN,-%2C%20joining%20sequences%20with

Reply:
But duckdb use `ASOF LEFT JOIN` https://duckdb.org/docs/guides/sql_features/asof_join.html. @st1page any insight on this?

Reply:
oops, then I have no preference.

Reply:
Both can be easily supported, so no need to block merging on this 😆 . We can always support the other syntax later.